libraries/common/subscriptions/router.spec.js

```diff
@@ -145,37 +162,37 @@ describe('Router subscriptions', () => {
         silent: true,
       };
 
-      callback(createCallbackPayload({ params }));
+      await callback(createCallbackPayload({ params }));
       testExpectedCall(conductor.push, params);
     });
 
-    it('should handle the ACTION_REPLACE history action as expected', () => {
+    it('should handle the ACTION_REPLACE history action as expected', async () => {
       const params = {
         action: ACTION_REPLACE,
         pathname: '/some_route',
         silent: false,
       };
 
-      callback(createCallbackPayload({ params }));
+      await callback(createCallbackPayload({ params }));
       testExpectedCall(conductor.replace, params);
     });
 
-    it('should remove trailing slashes from pathnames', () => {
+    it('should remove trailing slashes from pathnames', async () => {
       const params = {
         action: ACTION_PUSH,
         pathname: '/some_route/',
         state: {},
         silent: true,
       };
 
-      callback(createCallbackPayload({ params }));
+      await callback(createCallbackPayload({ params }));
       testExpectedCall(conductor.push, {
         ...params,
         pathname: params.pathname.slice(0, -1),
       });
     });
 
-    it('should redirect to a protector route when the user is not logged in', () => {
+    it('should redirect to a protector route when the user is not logged in', async () => {
```

Comment:
protected route

Reply:
`protector` is correct since the pushed route is protected. so the `navigate$` subscription redirects the user the the `protector` route e.g. login.